lib/index.js

```diff
@@ -282,17 +282,23 @@ CallsiteRecord.fromError = function (error, isCallsiteFrame) {
 };
 
 // API
-module.exports = function createCallsiteRecord (/* err, isCallsiteFrame || fnName, typeName */) {
-    if (arguments[0] instanceof Error)
-        return CallsiteRecord.fromError(arguments[0], arguments[1]);
+module.exports = function createCallsiteRecord (options) { /*{ forError, isCallsiteFrame, byFunctionName, typeName, processFrameFn }*/
+    if (options.forError)
+        return CallsiteRecord.fromError(options.forError, options.isCallsiteFrame);
 
-    else if (typeof arguments[0] === 'string') {
+    else if (options.byFunctionName) {
         var stackFrames = callsite();
 
+        if (options.processFrameFn) {
+            stackFrames = stackFrames.map(function (frame) {
```

Comment:
I agree, that this ability should be available, but processing should be executed before `CallsiteRecord` ctor